drgn_tools/iscsi.py

```diff
@@ -77,28 +82,34 @@ def for_each_iscsi_session(prog: Program) -> Iterator[Object]:
     :returns: a iterator of ``struct iscsi_session *``
 
     """
-    for h in for_each_iscsi_host(prog):
-        yield __get_iscsi_session(h)
+    for h in for_each_scsi_host(prog):
+        yield __get_iscsi_session(prog, h)
 
 
-def __get_iscsi_session(iscsi_host: Object) -> Object:
+def __get_iscsi_session(prog: Program, Scsi_Host: Object) -> Object:
     """
     Get the associated session given the scsi host.
     returns: ``struct iscsi_session *``
     """
-    return container_of(iscsi_host, "struct iscsi_session", "host")
+    iscsi_sw_tcp_host_addr = (
+        Scsi_Host.value_()
+        + sizeof(prog.type("struct Scsi_Host"))
+        + sizeof(prog.type("struct iscsi_host"))
+    )
+    iscsi_sw_tcp_host = Object(
+        prog, "struct iscsi_sw_tcp_host", address=iscsi_sw_tcp_host_addr
+    )
 
+    return iscsi_sw_tcp_host.session
```

Comment:
We should do some precheck to make sure this is sw tcp iscsi, if it's not, then leave a message saying it's not supported, otherwise the script will just crash for other iscsi type.